back-end/src/routes/api.js

```diff
@@ -341,9 +341,34 @@ const signup =  (request, response) => {
         }
       };
       try {
-        const fileData = await dynamodb.put(usersParams).promise();
-        statusCode = 200;
-        response.status(statusCode).send("signup successfully");
+        var queryUsers = {
+          TableName: "Users",
+          IndexName: "email-index",
+          KeyConditionExpression: "#email = :email",
+          ExpressionAttributeNames: {
+            "#email": "email"
+          },
+          ExpressionAttributeValues: {
+            ":email": email
+          }
+        };
+        var usersQueryData = await dynamodb.query(queryUsers).promise();       
+        if (usersQueryData.Items.length > 0){
+          const userInfo = {
+            userid: usersQueryData.Items[0].userid,
+            email: email
+          };
+          statusCode = 200;
```

Comment:
The code below is repeated twice in both `if` and `else` blocks.
```js
statusCode = 200;
response.status(statusCode).send(userInfo);
```
consider moving it out below the `else` block.